tensorflow_lite_support/python/test/task/vision/image_segmenter_test.py

```diff
@@ -196,6 +231,80 @@ def test_segment_model(self, model_file_type, expected_colored_labels):
             getattr(colored_labels[index], key),
             expected_colored_labels[index][key])
 
+  def test_segmentation_category_mask(self):
+    """Check if category mask matches with ground truth."""
+    # Creates segmenter.
+    base_options = _BaseOptions(file_name=self.model_path)
+    segmenter = _create_segmenter_from_options(
+      base_options, output_type=_OutputType.CATEGORY_MASK)
+
+    # Loads image.
+    image = tensor_image.TensorImage.create_from_file(self.test_image_path)
+
+    # Performs image segmentation on the input.
+    segmentation = segmenter.segment(image).segmentation[0]
+    result_pixels = np.array(bytearray(segmentation.category_mask))
+
+    # Loads ground truth segmentation file.
+    gt_segmentation = tensor_image.TensorImage.create_from_file(
+      self.test_seg_path)
+    gt_segmentation_array = gt_segmentation.buffer
+    gt_segmentation_shape = gt_segmentation_array.shape
+    num_pixels = gt_segmentation_shape[0] * gt_segmentation_shape[1]
+    ground_truth_pixels = gt_segmentation_array.flatten()
+
+    inconsistent_pixels = 0
+
+    for index in range(num_pixels):
+      inconsistent_pixels += (
+          result_pixels[index] * _MASK_MAGNIFICATION_FACTOR !=
+          ground_truth_pixels[index])
+
+    self.assertLessEqual(inconsistent_pixels / num_pixels,
+                         _MATCH_PIXELS_THRESHOLD,
+                         "Segmentation mask value must be the same size as ground truth.")
```

Comment:
The error message is incorrect here.

Besides, you should also assert if the segmentation mask size is the same as the ground truth mask size before the for loop.